npm/pkg/dataplane/policies/chain-management_linux_test.go

```diff
@@ -47,6 +47,50 @@ Chain AZURE-NPM-ACCEPT (1 references)
 `
 )
 
+func TestStaleChainsForceLock(t *testing.T) {
+	testChains := []string{}
+	for i := 0; i < 100000; i++ {
+		testChains = append(testChains, fmt.Sprintf("test-chain-%d", i))
+	}
+	calls := []testutils.TestCmd{}
+	for _, chain := range testChains {
+		calls = append(calls, getFakeDestroyCommand(chain))
+	}
+	ioshim := common.NewMockIOShim(calls)
+	// don't verify calls because there shouldn't be as many commands as we create if forceLock works properly
+	pMgr := NewPolicyManager(ioshim, ipsetConfig)
+
+	start := make(chan struct{}, 1)
+	done := make(chan struct{}, 1)
+	go func() {
+		pMgr.reconcileManager.Lock()
+		defer pMgr.reconcileManager.Unlock()
+		start <- struct{}{}
+		require.NoError(t, pMgr.cleanupChains(testChains))
```

Comment:
for cleanupChains function can we move https://github.com/Azure/azure-container-networking/blob/fea82838d9c7b8c23f20eaca9e8b16789e40a1e6/npm/pkg/dataplane/policies/chain-management_linux.go#L227-L237 to the `default` case in the select loop above it?